x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -1011,22 +1011,34 @@ private AbstractRunnable fillGapRunnable(
             @Nullable SourceInputStreamFactory streamFactory,
             ActionListener<Void> listener
         ) {
-            return ActionRunnable.run(listener.delegateResponse((l, e) -> failGapAndListener(gap, l, e)), () -> {
-                var ioRef = io;
-                assert regionOwners.get(ioRef) == CacheFileRegion.this;
-                assert CacheFileRegion.this.hasReferences() : CacheFileRegion.this;
-                int start = Math.toIntExact(gap.start());
-                writer.fillCacheRange(
-                    ioRef,
-                    start,
-                    streamFactory,
-                    start,
-                    Math.toIntExact(gap.end() - start),
-                    progress -> gap.onProgress(start + progress)
-                );
-                writeCount.increment();
-                gap.onCompletion();
-            });
+            return new AbstractRunnable() {
```

Comment:
can't we use `org.elasticsearch.action.ActionListener#run` here instead?

Reply:
I believe we needed `AbstractRunnable` for `gapFillingTasks` which is declared as `List<AbstractRunnable>`